runtime/interface.go

```diff
@@ -41,93 +43,124 @@ type Block struct {
 type ResolvedLocation = sema.ResolvedLocation
 type Identifier = ast.Identifier
 type Location = common.Location
-
-type Interface interface {
-	// ResolveLocation resolves an import location.
-	ResolveLocation(identifiers []Identifier, location Location) ([]ResolvedLocation, error)
-	// GetCode returns the code at a given location
-	GetCode(location Location) ([]byte, error)
-	// GetCachedProgram attempts to get a parsed program from a cache.
-	GetCachedProgram(Location) (*ast.Program, error)
-	// CacheProgram adds a parsed program to a cache.
-	CacheProgram(Location, *ast.Program) error
-	// GetValue gets a value for the given key in the storage, owned by the given account.
-	GetValue(owner, key []byte) (value []byte, err error)
-	// SetValue sets a value for the given key in the storage, owned by the given account.
-	SetValue(owner, key, value []byte) (err error)
-	// CreateAccount creates a new account.
-	CreateAccount(payer Address) (address Address, err error)
-	// AddAccountKey appends a key to an account.
-	AddAccountKey(address Address, publicKey []byte) error
-	// RemoveAccountKey removes a key from an account by index.
-	RemoveAccountKey(address Address, index int) (publicKey []byte, err error)
+type AddressLocation = common.AddressLocation
+
+// Accounts manages changes to the accounts, account keys and account storage
+//
+// errors returns by methods are non-fatal ones (e.g. accountNotExist),
+// fatal errors (e.g. Storage failure) are called by panic inside the methods
+// methods with power of edits have caller as an argument which can be used
+// to limit access to these method calls from smart contracts only, transaction only, or
+// specific smart contracts (e.g. service account).
+type Accounts interface {
+
+	// NewAccount creates a new account address and set the exists flag for this account
+	NewAccount(caller Location) (address Address, err error)
+	// Exists returns true if the account exists
+	Exists(address Address) (exists bool, err error)
+	// NumberOfAccounts returns the number of accounts
+	NumberOfAccounts(caller Address) (count uint64, err error)
+
+	// SuspendAccount suspends an account (set suspend flag to true)
+	SuspendAccount(address Address, caller Location) error
+	// UnsuspendAccount unsuspend an account (set suspend flag to false)
+	UnsuspendAccount(address Address, caller Location) error
+	// returns true if account is suspended
+	IsSuspended(address Address) (isSuspended bool, err error)
```

Comment:
Maybe keep naming consistent and include `Account` like in the other method names.

```suggestion
	IsAccountSuspended(address Address) (isSuspended bool, err error)
```

Reply:
done